agent-inject/agent/annotations.go

```diff
@@ -244,9 +245,22 @@ func Init(pod *corev1.Pod, cfg AgentConfig) error {
 	}
 
 	if _, ok := pod.ObjectMeta.Annotations[AnnotationAgentRunAsUser]; !ok {
+
 		if cfg.UserID == "" {
 			cfg.UserID = strconv.Itoa(DefaultAgentRunAsUser)
 		}
+		if cfg.SameID {
+			if len(pod.Spec.Containers) == 0 {
+				return errors.New("No containers found in Pod Spec")
+			}
+			if pod.Spec.Containers[0].SecurityContext == nil {
+				return errors.New("No SecurityContext found for Container 0")
+			}
+			if pod.Spec.Containers[0].SecurityContext.RunAsUser == nil {
+				return errors.New("RunAsUser is nil for Container 0's SecurityContext")
+			}
+			cfg.UserID = strconv.FormatInt(*pod.Spec.Containers[0].SecurityContext.RunAsUser, 10)
```

Comment:
I wonder if this should be looking at `pod.Spec.SecurityContext` as well.

Reply:
The `Spec.SecurityContext` is common for all containers running inside the pod. 
The `Spec.Container[0].SecurityContext` overrides the settings at the pod level.

Reply:
Yep, and after running this on OpenShift via CRC, I see that `RunAsUser` is being set at the container level.

I'm also now thinking that it might be simpler to just not set a SecurityContext at all on the init and sidecar containers, since OpenShift will do that for you based on the project/namespace.

Reply:
Mmm I have to double check it, but the reason that we explicitly put the security context is because we are injecting the side/init-container with the webhook and it is not OpenShift doing it.